docs/faq.md

```diff
@@ -1,21 +1,321 @@
+
+## What is the lifecycle of a function?
+
+Functions are packaged as Docker images and by default individual containers are created to handle a function request and are then destroyed.  However, `Hot Functions` are not disposed of after handling a single request.
+
+## What’s a `Hot Function`?
+
+`Hot Functions` are functions that are not destroyed after a single use but are retained and used to handle subsequent requests.  Hot Functions must accept HTTP input.
+
+# General
+
+## What is the Fn Project’s license?
+
+Apache 2.0
+
+## What is the URL of the project website?
+
+<http://fnproject.io>
+
+## Where is the source code hosted?
+
+All of the project code is on Github at <http://github.com/fnproject>.
+
+## Is Fn an entirely new platform?
+
+No, Fn is derived from the well-received IronFunctions project.  The core IronFunctions team are now at Oracle so Fn is simply the next evolutionary step of IronFunctions built by the original developers.
+
+## How do you pronounce “Fn”?
+
+Formally you can say “F" "N”, but in a sentence you might informally use “effen”.  For example, when describing how you invoke a function with the Fn CLI you might say: “effen call myapp myfunction”. ;-)
+
+# Network
+
+## Can Fn do IO over HTTP or in general any TCP based communication?
+
+Like most functions platforms we support HTTP.
+
+## Can an Fn function connect to various endpoints to read/write data?
+
+Yes, a function is not restricted in what it can connect to.
+
+## Any limits on the network bandwidth?
+
+We currently don't yet offer Fn as a managed service which would manage network traffic.  Currently, there are no restrictions in the core Fn platform.
+
+# Performance
+
+## How long can a function run? Can we make it run forever?
+
+We support 'hot functions' (see the end of [this tutorial](https://github.com/fnproject/tutorials/blob/master/JavaFDKIntroduction/README.md) for an example).  Hot functions will continue to live if they are used but, if not, will eventually be cleaned up.
+
+Function timeout is configurable. Please see <https://github.com/fnproject/fn/blob/master/docs/function-file.md>
+
+## How does the service trace the liveliness of the function? If my function dies/crashes will the service provision it again?
+
+This is not a microservices platform so the notion is slightly different.  Functions are run when called.  If you call a function that doesn't exist then one will be started.
+
+## How does scaling work? Should this be initiated by a customer?
+
+Fn runs more instances of the function based on requests.  It is automatic.
+
+## When one host is overloaded, will this service automatically relocate the function to another host?
+
+When running on Kubernetes (K8s) workloads are spread across available hosts.
+
+Functions currently do not relocate. Please see our [FnLB project inside Fn Project](https://github.com/fnproject/fn/tree/master/fnlb) for how to route requests to available hosts. K8s can also help with this.
+
+# Java
```

Comment:
Let's move this whole section into the https://github.com/fnproject/fdk-java repo, can create an faq.md there.

Reply:
Section Removed